### PR TITLE
fix: add reactive theme content

### PR DESCRIPTION
**Pull request by kaivanwong**
#### Description (required)

In `shiki`'s configuration, I set `experimentalThemes: { }` as documented, but it doesn't work when I dynamically switch dark mode and light mode.

image

output (1)

Then I checked the official documentation of `shiki` and found that I need to manually set some css fragments to get reactive dark mode.

image

I replaced the `.shiki` class name with `.astro-code` and added the code to my documentation project and now it works fine. 

output

I added this usage to the document, hoping to help developers who are unfamiliar with `shiki`. 😀

### CODE REVIEW

**Comment by Eveeifyeve** [comment]
!coauthor

**Comment by sarah11918** [comment]
Hi @kaivanwong ! This looks really cool!

We did just update our Shiki config though, and now instead of `experimentalThemes` it's just `themes`. Would you mind testing and make sure this still works with `themes`?

(See the updated, live docs now: https://docs.astro.build/en/guides/markdown-content/#shiki-configuration )

I will also ask @delucis to review, but we are a bit busy with launch week, so thank you for your patience as we catch up!  😄

**Comment by kaivanwong** [comment]
> Hi @kaivanwong ! This looks really cool!
> 
> We did just update our Shiki config though, and now instead of `experimentalThemes` it's just `themes`. Would you mind testing and make sure this still works with `themes`?
> 
> (See the updated, live docs now: [docs.astro.build/en/guides/markdown-content/#shiki-configuration](https://docs.astro.build/en/guides/markdown-content/#shiki-configuration) )
> 
> I will also ask @delucis to review, but we are a bit busy with launch week, so thank you for your patience as we catch up! 😄

Sure, I tested 'themes' and it work normaly, thank you. 

Also, it will be great if the experimental field names stay as the same. If I upgrade the version of Astro, I don't need to change my code. 😁

**Comment by Eveeifyeve** [comment]
> Hi @kaivanwong ! This looks really cool!
> 
> We did just update our Shiki config though, and now instead of `experimentalThemes` it's just `themes`. Would you mind testing and make sure this still works with `themes`?
> 
> (See the updated, live docs now: https://docs.astro.build/en/guides/markdown-content/#shiki-configuration )
> 
> I will also ask @delucis to review, but we are a bit busy with launch week, so thank you for your patience as we catch up! 😄

Yeah I thought this LGTM Since I would use this feature as well.

**Comment by sarah11918** [comment]
Hi @kaivanwong ! We're all catching up now after a busy launch week, and I just noticed the second file changed in this PR, too.

Can you please remove the `zh` file in this PR? When content changes are proposed to the English site, the PR should contain only a change to the English file, which is treated as the source file for our i18n tracking system.

Only after the English is reviewed, edited, and approved do we consider changes ready for translation. Even small changes to the English site are often revised before publishing, so we ask that you not worry about submitting a translation at the same time, to save you time, and because we have a system for translations already! 🙌

**Comment by kaivanwong** [comment]
> Hi @kaivanwong ! We're all catching up now after a busy launch week, and I just noticed the second file changed in this PR, too.
> 
> Can you please remove the `zh` file in this PR? When content changes are proposed to the English site, the PR should contain only a change to the English file, which is treated as the source file for our i18n tracking system.
> 
> Only after the English is reviewed, edited, and approved do we consider changes ready for translation. Even small changes to the English site are often revised before publishing, so we ask that you not worry about submitting a translation at the same time, to save you time, and because we have a system for translations already! 🙌

ok, it doesn't matter, I've seen a major update happening to Astro recently. Now, I droped the commit for the `zh` file.

**Review by sarah11918 — COMMENTED** [review]
Hi @kaivanwong , thank you for your patience while we discussed this proposed change internally!

We are glad you added the extra information about the `astro-code` class, and think that's really helpful to have here. We have decided that the exact implementation should be left to Shiki's own docs, since they do have this example there and will keep it updated and maintained. We try whenever possible not to document anything that might change without us knowing if it already exists somewhere else and we can just link to it.

I think adding just a note, and not the specific implementation, calls more attention to Astro's class name, which is really the key idea to get across!

So see what you think of this suggestion!

File: src/content/docs/en/guides/markdown-content.mdx [hunk 1]
```diff
@@ -613,6 +613,39 @@ export default defineConfig({
...
+  background-color: var(--shiki-dark-bg) !important;
+  /* Optional, if you also want font styles */
+  font-style: var(--shiki-dark-font-style) !important;
+  font-weight: var(--shiki-dark-font-weight) !important;
+  text-decoration: var(--shiki-dark-text-decoration) !important;
+}
+```
+
+> The class name used in [shiki](https://shiki.style/guide/dual-themes#query-based-dark-mode) is `.shiki`, here you need to use `.astro-code`.
+
```
[the remark this answers is not in the document]
Comment (kaivanwong) [reply src/content/docs/en/guides/markdown-content.mdx]: 😀 Thank you very much for your and I think you are right.

**Review by sarah11918 — APPROVED** [review]
Thank you @kaivanwong ! I think this note is going to be really helpful, and really call attention to the main point. Thank you for taking the time to contribute this, and welcome to Team Docs! :partying_face:

**Comment by kaivanwong** [comment]
> Thank you @kaivanwong ! I think this note is going to be really helpful, and really call attention to the main point. Thank you for taking the time to contribute this, and welcome to Team Docs! :partying_face:

Thank you, and I am happy to be able to make some small contribution. 😸

**Pull request merged**